File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    let count = 0;
    const arr = new Array(n+1).fill(1);
    
    for(let i = 2; i <= n; i++) {
        if(arr[i] === 0) continue;
        
        for(let j = i*2; j <= n; j+=i){
            arr[j] = 0;
        }
    }
    
    return arr.filter(a => a === 1).length -2;
}

const isPrime = num => {
    for(let i = 2; i <= Math.sqrt(num); i++) {
        if( num % i === 0 && i !== num) return false;
    }
    return true;
}

/*
소수: 1
소수가 아닌 수: 0
*/

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    arr = [True for _ in range(n + 1)]
    arr[0] = False
    arr[1] = False
    
    prev = 2
    while(True):
        if arr[prev] == False:
            for i in range(prev + 1, len(arr)):
                if arr[i] == True:
                    prev = i
                    break
                    
        if prev ** 2 > n:
            break
            
        for i in range(prev * 2, len(arr), prev):
            arr[i] = False
            
        prev += 1
        
    s = set()
    for i in range(len(arr)):
        if arr[i] == True:
            s.add(i)
            
    return len(s)
            
        
    
        
    
    
